types/2020-08-27/SubscriptionItems.d.ts

```diff
@@ -261,6 +300,33 @@ declare module 'stripe' {
       }
 
       type ProrationBehavior = 'always_invoice' | 'create_prorations' | 'none';
+
+      interface Trial {
+        free?: Trial.Free;
+
+        none?: Trial.None;
+
+        /**
+         * Details of a different price, quantity, or both, to bill your customer for during a paid trial.
+         */
+        paid?: Trial.Paid;
+
+        type: Trial.Type;
+      }
+
+      namespace Trial {
+        interface Free {}
```

Comment:
This is weird too

Reply:
This is expected - it's a polymorphic value where based on the type, you'll have some extra type-specific properties (eg. see the `Paid` interface below). Basically this is a "one of".